packages/devtools_app/lib/src/debugger/variables.dart

```diff
@@ -72,16 +72,19 @@ class VariablesList extends StatelessWidget {
 }
 
 class ExpandableVariable extends StatelessWidget {
-  const ExpandableVariable({Key key, this.variable}) : super(key: key);
+  const ExpandableVariable({Key key, this.variable, this.debuggerController})
+      : super(key: key);
 
   final ValueListenable<Variable> variable;
+  final DebuggerController debuggerController;
 
   @override
   Widget build(BuildContext context) {
     return ValueListenableBuilder<Variable>(
       valueListenable: variable,
       builder: (context, variable, _) {
-        final controller = Provider.of<DebuggerController>(context);
+        final controller =
+            debuggerController ?? Provider.of<DebuggerController>(context);
```

Comment:
Is this because we don't have a debugger controller available in the context of the hover card overlay?

Reply:
correct. Overlay doesn't play great with provider. We might be able to resolve if we moved our providers to the absolute root of the tree but that would cause problems for the snapshot app case.